reposerver/repository/repository.proto

```diff
@@ -28,6 +28,19 @@ message ManifestRequest {
     bool verifySignature = 16;
 }
 
+// TestRepositoryRequest is a query to test repository is valid or not and has valid access.
+message TestRepositoryRequest {
+    github.com.argoproj.argo_cd.pkg.apis.application.v1alpha1.Repository repo = 1;
+    bool isHelm = 2;
```

Comment:
Nitpick: rather than two booleans, maybe this should be a `type` string: `git`, `helm`, `helmOCI`, if that makes sense?